### ISSUE TITLE
[terra-menu-item] classnames not copied when cloning icon

### ISSUE THREAD

**Issue by kschuste** [cerner/terra-framework#1599]
# Bug Report

## Description
The terra-menu now allows icons to be placed in the menu.  This includes any icons, in particular, the themeable icons.  However, if any styling is applied to the icon via classnames, those styles will not get carried forward.

## Steps to Reproduce
1.  Create a terra-menu with menu-items that contain icons with icons that can be themed
2.  Apply a classname to the icon that will apply styling to it
3.  Run test and verify in the menu that the classname styling is not applied
4.  Can verify that the classname is missing in the inspector

## Additional Context / Screenshots
Required icon (asterisk) is missing classname styling to make it rec

## Expected Behavior
Expected behavior is to carry forward the classname styling and apply the red coloring
![default_icon](https://user-images.githubusercontent.com/1711642/218533579-99698bf4-22b2-4273-8807-3019c142450c.png)

## Possible Solution
When cloning the icon in the MenuItem.jsx, append the start-icon styling to any existing classnames.

## Environment
This can be found in any environment

## @ Mentions
@neilpfeiffer @supreethmr @BenBoersma
